client/app/bundles/comments/components/CommentBox/CommentList/CommentList.jsx

```diff
@@ -20,57 +20,68 @@ export default class CommentList extends BaseComponent {
 
   constructor(props, context) {
     super(props, context);
-    this.state = {};
+
     _.bindAll(this, 'errorWarning');
   }
 
   errorWarning() {
+    const { error, cssTransitionGroupClassNames } = this.props;
+
     // If there is no error, there is nothing to add to the DOM
-    if (!this.props.error) return null;
+    if (!error) return null;
+
+    const nodeRef = React.createRef(null);
+
     return (
-      <Alert bsStyle="danger" key="commentFetchError">
-        <strong>Comments could not be retrieved. </strong>A server error prevented loading comments. Please
-        try again.
-      </Alert>
+      <CSSTransition
+        key="commentFetchError"
+        nodeRef={nodeRef}
+        timeout={500}
+        classNames={cssTransitionGroupClassNames}
+      >
+        <Alert ref={nodeRef} bsStyle="danger">
+          <strong>Comments could not be retrieved. </strong>A server error prevented loading comments. Please
+          try again.
+        </Alert>
+      </CSSTransition>
     );
   }
 
   render() {
     const { $$comments, cssTransitionGroupClassNames } = this.props;
-    const commentNodes = $$comments.map(($$comment, index) => (
-      // `key` is a React-specific concept and is not mandatory for the
-      // purpose of this tutorial. if you're curious, see more here:
-      // http://facebook.github.io/react/docs/multiple-components.html#dynamic-children
-      <Comment
-        key={$$comment.get('id') || index}
-        author={$$comment.get('author')}
-        text={$$comment.get('text')}
-      />
-    ));
+    const commentNodes = $$comments.map(($$comment, index) => {
+      const nodeRef = React.createRef(null);
+      return (
+        // `key` is a React-specific concept and is not mandatory for the
+        // purpose of this tutorial. if you're curious, see more here:
+        // http://facebook.github.io/react/docs/multiple-components.html#dynamic-children
```

Comment:
Any React developer should know to include unique `key` to lists to avoid issues when interacting with them. We don't need to mention it here.